File: demo/internal/model/Dispute.go
package model

import (
    "time"
    "gorm.io/gorm"
)

//==============================================================
// Dispute Declaration
//==============================================================
type Dispute struct {
    gorm.Model
     DisputeReference                                    string
    RaisedOn                                                            time.Time
    Reason                                    string
    TransactionId         *uint
    Transaction           *Transaction `gorm:"foreignKey:TransactionId"`
    CustomerId         *uint
    Customer           *Customer `gorm:"foreignKey:CustomerId"`
    AccountId         *uint
    Account           *Account `gorm:"foreignKey:AccountId"`
    PaymentCardId         *uint
    PaymentCard           *PaymentCard `gorm:"foreignKey:PaymentCardId"`
    Status                      DisputeStatus

// parent associations as their child

}

File: bankingOnDjango/models/Dispute.py
from django.db import models
from bankingOnDjango.models.DisputeStatus import DisputeStatus


# ======================================================================
# Class Dispute Declaration
# ======================================================================
class Dispute(models.Model):

    # ======================================================================
    # attribute declarations
    # ======================================================================
    dispute_reference = models.CharField(max_length=200, null=True)
    raised_on = models.DateField(null=True)
    reason = models.CharField(max_length=200, null=True)
    transaction = models.ForeignKey(
        "Transaction", on_delete=models.CASCADE, null=True, blank=True, related_name="+"
    )
    customer = models.ForeignKey(
        "Customer", on_delete=models.CASCADE, null=True, blank=True, related_name="+"
    )
    account = models.ForeignKey(
        "Account", on_delete=models.CASCADE, null=True, blank=True, related_name="+"
    )
    payment_card = models.ForeignKey(
        "PaymentCard", on_delete=models.CASCADE, null=True, blank=True, related_name="+"
    )
    status = models.CharField(
        max_length=64,
        null=True,
        choices=[(tag.name, tag.value) for tag in DisputeStatus],
    )

    # ======================================================================
    # function declarations
    # ======================================================================
    def toString(self):
        str = ""
        str = str + self.disputeReference
        str = str + self.raisedOn
        str = str + self.reason
        str = str + self.status
        return str

    def __str__(self):
        return self.toString()

    def identity(self):
        return "Dispute"

    def objectType(self):
        return "Dispute"
